### PR TITLE
Declare an argument parser for each command

### PR DESCRIPTION
**Pull request by 00vareladavid**
This changes the command declarations.

For example, the `pin` command: 
`(ARG_VERSION, [])` -> `([], parse_pkg, [:valid => [VersionRange]]),`

The first field lists the number of expected arguments **after** they have been parsed (empty means don't count).
The next field specifies the parser to be applied on that command's arguments.
The final field allows a command to key into a parser.

### CODE REVIEW

File: src/REPLMode.jl [hunk 1]
```diff
@@ -901,7 +893,7 @@ command_declarations = [
     CMD_REGISTRY_ADD,
     ["add"],
     do_registry_add!,
-    (ARG_PKG, []),
+    ([], nothing, []),
```
Comment (fredrikekre) [inline src/REPLMode.jl]: We can also use `identity` as a parser and change `Union{Function,Nothing}` with judt `Function`.

**Comment by 00vareladavid** [comment]
Updates for 1.0 from @fredrikekre's #588

**Comment by KristofferC** [comment]
Argh, we should have squashed this.

**Pull request merged**